packages/ui/Slottable/Slottable.js

```diff
@@ -103,16 +43,32 @@ const defaultConfig = {
  */
 const Slottable = hoc(defaultConfig, (config, Wrapped) => {
 	const slots = config.slots;
-	return kind({
-		name: 'Slottable',
-		render: (props) => {
-			const adjusted = distribute(slots, props);
-			return (
-				<Wrapped {...adjusted} />
-			);
-		}
-	});
+
+	// eslint-disable-next-line no-shadow
+	return function Slottable ({children, ...rest}) {
+		const distributed = useSlots({slots, children});
+
+		// Slottable allows there to be other values in the destination slot and merges them.
+		// However, consumers can't avoid key warnings when merging the two lists so we should
+		// "consider this harmful" and not continue to support this with the hook and instead
+		// encourage "either/or". We should continue to support it here so this block ensures
+		// backwards compatibility.
+		slots.forEach(slot => {
+			const dist = distributed[slot];
+			const prop = rest[slot];
+			if (prop && dist) {
+				distributed[slot] = Array.isArray(dist) ? [prop, ...dist] : [prop, dist];
```

Comment:
Do we need to check if `prop` is an array as well?  Also, should we add any tests around this functionality?

Reply:
Maybe `distributed[slot] = [].concat(prop, dist)`?

Reply:
Maybe it's fine as-is.  Technically `dist` could have been a single slot that contains an array.  It could be a bit difficult to figure out.

Reply:
The current code does seem to handle this scenario (surprisingly to me). I'll push an update!

Reply:
Thanks for the tip, @webOS101! Fixed using your suggestion.